lib/spack/spack/cmd/install.py

```diff
@@ -332,19 +342,43 @@ def install(parser, args, **kwargs):
             spec, test_suite
         )(PackageBase.do_install)
 
-    # Do the actual installation
-    if args.things_to_install == 'dependencies':
+    yield
+
+    # Dump log file if asked to
+    if args.log_format is not None:
+        test_suite.dump(log_filename)
+
+def top_install(spec, install_package=True, install_dependencies=True, **kwargs):
+    """Top-level install method."""
+    if install_dependencies:
         # Install dependencies as-if they were installed
         # for root (explicit=False in the DB)
-        kwargs['explicit'] = False
         for s in spec.dependencies():
-            p = spack.repo.get(s)
-            p.do_install(**kwargs)
-    else:
+            package = spack.repo.get(s)
+            package.do_install(install_dependencies=True, explicit=False, **kwargs)
+
+    if install_package:
```

Comment:
Not a big deal, but why we changed the semantics here? We are driving dependencies installation explicitly when it shouldn't be necessary: `do_install` knows how to recurse internally and here we are mixing two competing recursions on dependencies.